indico/modules/events/management/controllers/protection.py

```diff
@@ -71,34 +101,85 @@ class RHEventProtection(RHManageEventBase):
                                  if p.has_management_permission('registration', explicit=True)}
         event_session_settings = session_settings.get_all(self.event)
         coordinator_privs = {name: event_session_settings[val] for name, val in COORDINATOR_PRIV_SETTINGS.iteritems()
                              if event_session_settings.get(val)}
-        return dict({'protection_mode': self.event.protection_mode, 'acl': acl, 'managers': managers,
-                     'registration_managers': registration_managers, 'submitters': submitters,
+        permissions = [[serialize_principal(p.principal), list(self._get_principal_permissions(p))]
+                       for p in self.event.acl_entries if self._get_principal_permissions(p)]
+
+        return dict({'protection_mode': self.event.protection_mode, 'registration_managers': registration_managers,
                      'access_key': self.event.access_key, 'visibility': self.event.visibility,
-                     'own_no_access_contact': self.event.own_no_access_contact}, **coordinator_privs)
+                     'own_no_access_contact': self.event.own_no_access_contact, 'permissions': permissions},
+                    **coordinator_privs)
 
     def _update_session_coordinator_privs(self, form):
         data = {field: getattr(form, field).data for field in form.priv_fields}
         update_session_coordinator_privs(self.event, data)
+
+    def _get_principal_permissions(self, principal):
+        """Retrieve a set containing the valid permissions of a principal."""
+        permissions = set()
+        if principal.full_access:
+            permissions.add(FULL_ACCESS_PERMISSION)
+        if principal.read_access:
+            permissions.add(READ_ACCESS_PERMISSION)
+        available_permissions = get_permissions_info()[0]
+        return permissions | (set(principal.permissions) & set(available_permissions))
+
+    def _update_permissions(self, current, new):
+        """Handle the updates of permissions and creations/deletions of acl principals.
+
+        :param current: A dict mapping principals and a set with its current permissions
+        :param new: A dict mapping principals and a set with its new permissions
+        """
+        def _get_splitted_permissions(_permissions):
+            full_access_permission = FULL_ACCESS_PERMISSION in _permissions
+            if full_access_permission:
+                _permissions.remove(FULL_ACCESS_PERMISSION)
+            read_access_permission = READ_ACCESS_PERMISSION in _permissions
+            if read_access_permission:
+                _permissions.remove(READ_ACCESS_PERMISSION)
+            return full_access_permission, read_access_permission, _permissions
+
+        for principal, permissions in current.viewitems():
+            if principal not in new:
+                self.event.update_principal(principal, False, False, del_permissions=current[principal])
```

Comment:
Anything besides the first argument should be a kwarg.

Can't wait for Python 3 to enforce stuff like this! ;)

Reply:
Also, you are removing someone's access here, we have `remove_principal()` for exactly this purpose. Just FYI, if you really needed to use `update_principal()` you could just pass `permissions=set()` to remove them all instead of explicitly specifying what to remove...

Reply:
The problem is that this user can have other permissions not involved in the Protection page like for instance `paper_layout_reviewer` that we don't want to remove.

Reply:
ok, valid point. maybe we could build a list containing such permissions as well? something like this:

    new_permissions = (existing_permissions & not_user_editable_permissions)

Like this everything that's not exposed in the permissions field is kept.

In the case below (updating instead of removing), you could do something like this:

    new_permissions = (existing_permissions & not_user_editable_permissions) | new_permissions